static/js/util.js

```diff
@@ -340,9 +345,12 @@ export function filter_by_word_prefix_match(items, search_term, item_to_text) {
     const filtered_items = items.filter((item) =>
         search_terms.some((search_term) => {
             const lower_name = item_to_text(item).toLowerCase();
-            const cands = lower_name.split(" ");
-            cands.push(lower_name);
-            return cands.some((name) => name.startsWith(search_term));
+            // returns true if the item starts with the search term or if the
+            // search term with a word separator right before it appears in the item
+            return (
+                lower_name.startsWith(search_term) ||
+                new RegExp(word_separator_regex.source + search_term).test(lower_name)
```

Comment:
So, generally it's not safe to combine something into a regular expression via just `+`; one would need to escape special regular expression characters like `.` for this to be correct, or they'll be treated as part of the regular expression itself.

I don't think we're intending this to feel like an "enter a regular expression" UI, so we should escape the `search_term` before doing this concatenation, I think? 

I'm not entirely sure what you have in mind, but a good test case is searching for strings making use of `.` or `*` or `?`.

Reply:
Thanks for pointing out this edge case. As you suspected, things do not work as expected when using those special characters. 

Here "test" should not show up when searching for `*st` but it does since the search term is (unintentionally) treated as a regular expression:
![image](https://user-images.githubusercontent.com/68962290/180923137-3a1790ee-3fe7-46e7-b7ef-5c2c9cfea830.png)

Escaping the search term fixes this!